### DISCUSSION TITLE
connect 2 form in 1 form

### DISCUSSION THREAD

**Discussion by samaremad** [serenity-is/Serenity#5572] [Help]
hi everyone how are you?

i have 2 row  each one of them have a link in side menu as you see

![Screenshot (96)](https://user-images.githubusercontent.com/77009503/110443457-995ebc00-80c4-11eb-8bce-c1e7993bd5d2.png)

![Screenshot (97)](https://user-images.githubusercontent.com/77009503/110443461-9a8fe900-80c4-11eb-9b3c-2dcfdbcbee32.png)

i want to remove voting link from menu and when i open group link and press on new group or edit existing records open form contain (form of group data and an icon when i press on it show the voting data that related to that group)

![Screenshot (99)](https://user-images.githubusercontent.com/77009503/110443932-230e8980-80c5-11eb-9c76-ddf677bcdd4c.png)

![Screenshot (100)](https://user-images.githubusercontent.com/77009503/110443946-26097a00-80c5-11eb-8fea-c1d282593b09.png)

in another word i want both these forms show in one form (group form )

does anyone have an idea about how to do that?

**Comment by luongnv6** [comment]
```C#
protected getColumns() {
            var columns = super.getColumns();

            Q.first(columns, x => x.field == 'GroupName').format = ctx => {
                let item: GroupRow = ctx.item;
                return `<a href="#" class="view-edit-group">${item.GroupName}</a>`;
            }

            return columns;
        }

        protected onClick(e: JQueryEventObject, row: number, cell: number) {
            super.onClick(e, row, cell);

            if (e.isDefaultPrevented())
                return;

            var item = this.itemAt(row);
            var target = $(e.target);

            if (target.hasClass('view-edit-group')) {
                 let dlg = new GroupDialog();
                this.initDialog(dlg);
                dlg.loadByIdAndOpenDialog(item.GroupId);
            }
        }
```

hope help you

**Reply by samaremad** [reply]
i will added this code in group dialogue, right?

**Reply by luongnv6** [reply]
i'm sorry. it is VotingGrid

**Reply by samaremad** [reply]
thank you so much i will try it now and change the names and see what can i reach and tell you

**Reply by samaremad** [reply]
![Screenshot (101)](https://user-images.githubusercontent.com/77009503/110593367-261c7f00-8184-11eb-95c7-58af988b3aac.png)

i get this error 

and i edit code as the following 


`
protected getColumns() {
            var columns = super.getColumns();

            Q.first(columns, x => x.field == 'Title').format = ctx => {
                let item: VotingRow = ctx.item;
                return `<a href="#" class="view-edit-Voting">${item.Title}</a>`;
            }

            return columns;
        }

        protected onClick(e: JQueryEventObject, row: number, cell: number) {
            super.onClick(e, row, cell);

            if (e.isDefaultPrevented())
                return;

            var item = this.itemAt(row);
            var target = $(e.target);

            if (target.hasClass('view-edit-Voting')) {
                let dlg = new VotingDialog();
                this.initDialog(dlg);
                dlg.loadByIdAndOpenDialog(item.Id);
            }`

**Comment by samaremad** [comment]
![Untitled](https://user-images.githubusercontent.com/77009503/110595643-f0c56080-8186-11eb-9359-ee2ad686f854.png)

this is a figure to describe what i need to do  
this 1 form 
but when you click on new icon the form will be like the figure in the right 
how can i do that please help me
